homeassistant/components/solaredge/sensor.py

```diff
@@ -426,20 +426,24 @@ def update(self):
         self.unit = energy_details["unit"]
         meters = energy_details["meters"]
 
-        for entity in meters:
-            for key, data in entity.items():
-                if key == "type" and data in [
-                    "Production",
-                    "SelfConsumption",
-                    "FeedIn",
-                    "Purchased",
-                    "Consumption",
-                ]:
-                    energy_type = data
-                if key == "values":
-                    for row in data:
-                        self.data[energy_type] = row["value"]
-                        self.attributes[energy_type] = {"date": row["date"]}
+        try:
+            for entity in meters:
+                for key, data in entity.items():
+                    if key == "type" and data in [
+                        "Production",
+                        "SelfConsumption",
+                        "FeedIn",
+                        "Purchased",
+                        "Consumption",
+                    ]:
+                        energy_type = data
+                    if key == "values":
+                        for row in data:
+                            self.data[energy_type] = row["value"]
+                            self.attributes[energy_type] = {"date": row["date"]}
```

Comment:
The only places that can raise KeyError are these two lines. We shouldn't wrap all code.